apps/client/src/hooks/use-roles.ts

```diff
@@ -0,0 +1,11 @@
+import { getRoles, getRolesUrl } from "@/services/roles";
+import { Role } from "@/types";
+import useSWR from "swr";
+
+export function useRoles(search?: string) {
+  const url = getRolesUrl(search);
+
+  return useSWR<Role[]>(url, () => getRoles(search), {
+    keepPreviousData: true,
```

Comment:
When navigating between pages, this option allows the previous (potentially stale data) to be visible while fetching fresh data.

I felt this was a much better user experience than showing a skeleton load indicator every time.